### PR TITLE
Device reset if no comm activity for while.

### LINKED ISSUE #128: Device disconnected message
**Is your feature request related to a problem? Please describe.**
If MaxMix software is not launched (or has crashed) the device continue to behave normally, but it is not doing anything

**Describe the solution you'd like**
When not asleep, the device's display could show a disconnected message.

### PR DESCRIPTION
**Pull request by ItsLemmy**
## Issues
 - Fixes #128 

## Description
The host master app send a heartbeat message every 5sec.
The slave device reset itself if no message received for 15sec.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] Requested changes are in a branch
- [x] Compiled and tested requested changes on target hardware (PC, device)
- [ ] Updated the documentation, if necessary
- [ ] Updated the LICENSES file, if necessary
- [x] Reviewed the [guidelines for contributing](../CONTRIBUTING.md) to this repository

### CODE REVIEW

File: Desktop/Application/MaxMix/Services/Communication/CommunicationService.cs [hunk 1]
```diff
@@ -200,6 +203,16 @@ private void HandleReconnection()
                         }
                     }
                 }
+                else
+                {
+                    // Send a regular heartbeat
+                    if (_heartbeatWatch.Elapsed.TotalMilliseconds > 5000)
```
Comment (t3knomanzer) [inline Desktop/Application/MaxMix/Services/Communication/CommunicationService.cs]: Extract this to a constant?

Comment (t3knomanzer) [reply Desktop/Application/MaxMix/Services/Communication/CommunicationService.cs]: Should we reduce the hearbeat time? 5s seems a bit excesive, doesn't it?

File: Desktop/Application/MaxMix/Services/Communication/Messages/MessageHeartbeat.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+namespace MaxMix.Services.Communication.Messages
+{
+    internal class MessageHeartbeat : IMessage
+    {
+        #region Constructor
+        public MessageHeartbeat() { }
+        #endregion
+
+        #region Properties
+        public byte Revision { get; private set; }
```
Comment (t3knomanzer) [inline Desktop/Application/MaxMix/Services/Communication/Messages/MessageHeartbeat.cs]: This is unused, right?

File: Embedded/MaxMix/MaxMix.ino [hunk 3]
```diff
@@ -178,8 +185,12 @@ void loop()
     UpdateLighting();
   }
 
-  Display::UpdateTimers(now - last);
-  isDirty = false;
+  // Reset / Disconnect if no serial activity.
+  if (lastCommTime + 15000 < now) {
```
Comment (t3knomanzer) [inline Embedded/MaxMix/MaxMix.ino]: Can we put the 15000 into a const and maybe reduce the the time to 2x the hearbeat time? 15s seems like a lot!

**Review by t3knomanzer — CHANGES_REQUESTED** [review]

**Comment by t3knomanzer** [comment]
The device seems to wake up every 15s if left disconnected in the splash screen.

File: Desktop/Application/MaxMix/Services/Communication/Messages/MessageHeartbeat.cs (hunk 2 above)
Comment (ItsLemmy) [reply Desktop/Application/MaxMix/Services/Communication/Messages/MessageHeartbeat.cs]: right

File: Desktop/Application/MaxMix/Services/Communication/CommunicationService.cs (hunk 1 above)
Comment (ItsLemmy) [reply Desktop/Application/MaxMix/Services/Communication/CommunicationService.cs]: set it to 2sec

File: Embedded/MaxMix/MaxMix.ino (hunk 3 above)
Comment (ItsLemmy) [reply Embedded/MaxMix/MaxMix.ino]: Set it in var to 4sec. 
One will need to raise it up quite a bit, if he need to use breakpoints in the desktop app.

**Comment by t3knomanzer** [comment]
When disconnected, the leds play the spinning animation but the display doesn't go to the splashScreen until you rotate or press the knob to cause a displayUpdate

**Comment by ItsLemmy** [comment]
Yeah that's the trade off I came up with to avoid burning in the display if it stays disconnected for a very long time.

**Comment by t3knomanzer** [comment]
It’s a different thing, what I’m referring to is a bug.

https://imgur.com/a/hUQa1sp

**Review by t3knomanzer — APPROVED** [review]

**Pull request merged**